### ISSUE TITLE
I want to use calculation modifier for data placeholders

### ISSUE THREAD

**Issue by takutok** [amcharts/amcharts5#348]
**Question**

I'm migrating code from a previous version.
I want to display the average value and maximum value data in Legend.
Modifies were used for data placeholders  in past versions, but isn't it supported in V5?

V4
https://www.amcharts.com/docs/v3/reference/amlegend/#periodValueText_property
V3
https://www.amcharts.com/docs/v4/concepts/formatters/formatting-strings/

**Comment by martynasma** [comment]
Well, it works differently in amCharts 5. You can access related data item's settings.

You can refer to any series data item setting, as it is described here:
https://www.amcharts.com/docs/v5/charts/xy-chart/legend-xy-series/#Dynamic_values

To set up value labels refer to this tutorial:
https://www.amcharts.com/docs/v5/charts/xy-chart/legend-xy-series/#Value_labels

**Comment by takutok** [comment]
Thankyou, I know how to use those Legend features.
In the previous version, the maximum value, minimum value, average value, etc. of the data were calculated automatically.
It was very useful. If in V5 they aren't available, it's back.

**Comment by takutok** [comment]
Thank you.
I found out why the Legend value couldn't be displayed.
Legend seems to be empty when all the data is 0.

![legend](https://user-images.githubusercontent.com/1034808/160263010-1ea5ac57-4eea-41e9-a5f2-b9d899bc75a1.png)

```
var data = [{
  date: new Date(2021, 0, 1).getTime(),
  value: 0,
  value2: 2.5
}, {
  date: new Date(2021, 0, 2).getTime(),
  value: 0,
  value2: 2.1
}, {
  date: new Date(2021, 0, 3).getTime(),
  value: 0,
  value2: 3
}, {
  date: new Date(2021, 0, 4).getTime(),
  value: 0,
  value2: 2
}, {
  date: new Date(2021, 0, 5).getTime(),
  value: 0,
  value2: 0.5
}, {
  date: new Date(2021, 0, 6).getTime(),
  value: 0,
  value2: 1.5
}, {
  date: new Date(2021, 0, 7).getTime(),
  value: 0,
  value2: 1.8
}];

// Create series
function createSeries(name, field) {
  var series = chart.series.push( 
    am5xy.LineSeries.new(root, { 
      name: name,
      xAxis: xAxis, 
      yAxis: yAxis, 
      valueYField: field, 
      valueXField: "date",
      calculateAggregates: true,
      tooltip: am5.Tooltip.new(root, {}),
      legendLabelText: "[bold {stroke}]{name}:[/]",
      legendRangeLabelText: "[{stroke}]{name}:[/]",
      legendValueText: "[bold {stroke}]{valueYHigh}[/]",
      legendRangeValueText: "[{stroke}]{valueYHigh}[/]"
    }) 
  );
```
I tried using Formatting Numbers, but I couldn't set the empty data to 0.
I also want to display 0 in Legend. Please give me a solution.

**Comment by martynasma** [comment]
> Legend seems to be empty when all the data is 0.

It's a bug. Will fix it in the next version and will let you know when it's out.

**Comment by martynasma** [comment]
Fixed in [5.1.10](https://github.com/amcharts/amcharts5/releases/tag/5.1.10).

## [5.1.10] - 2022-03-30

### Added
- New in-line text formatting property `verticalAlign` added. Currently supports only `"super"` and `"sub"` values. E.g. `Copyright[fontSize: 8px; verticalAlign: super;]TM[\]`.

### Changed
- Logarithmic scale on `ValueAxis` was revamped with a different approach. No changes from the configration, but might influence appearance/scale.
- When calculating derivative values for series, `low`, `high`, and `open` used previous `low`, `high`, and `open` values. Now they all use `value` instead.
- Updated to D3 v7.

### Fixed
- In some cases, using a text placehodler, an empty string was shown insteaf of a zero.
- Some grid container masking issues fixed.
- Sometimes `valueAxis` was zoomed-in a bit on the first render of an `XYChart`.
- Axis ranges with `dRadius > 0` were not visible on `RadarChart`.
- Dynamic change of `value*Show` fields of an `XYSeries` was not working.
- Removing data from a legend could lead to JS errors in some particular cases.
- Ghost label was creating a gap in an `XYChartScrollbar` between scrollbar bottom an scrollbar series.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
